rest/route/service.go

```diff
@@ -158,6 +158,7 @@ func AttachHandler(app *gimlet.APIApp, opts HandlerOpts) {
 	app.AddRoute("/projects/{project_id}/versions/tasks").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeFetchProjectTasks(sc))
 	app.AddRoute("/projects/{project_id}/patch_trigger_aliases").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeFetchPatchTriggerAliases(sc))
 	app.AddRoute("/projects/{project_id}/parameters").Version(2).Get().Wrap(checkUser, viewTasks).RouteHandler(makeFetchParameters(sc))
+	app.AddRoute("/projects/variables/rotate").Version(2).Put().Wrap(checkUser, createProject).RouteHandler(makeProjectVarsPut(sc, env.Settings()))
```

Comment:
this route doesn't use settings, so your function shouldn't take it in

Reply:
done